pkg/pool/pool.go

```diff
@@ -70,31 +73,42 @@ type pool struct {
 	// reSyncRequired is set if the upstream and pool are possibly out of sync due to
 	// errors in creating/deleting resources
 	reSyncRequired bool
+	// isPDPool indicates whether the pool is for prefix IP provider or secondary IP provider
+	isPDPool bool
+}
+
+// Resource represents a secondary IPv4 address or a prefix-deconstructed IPv4 address, uniquely identified by GroupID and ResourceID
+type Resource struct {
+	// could be IPv4 address or IPv4 prefix
+	GroupID string
+	// IPv4 address
+	ResourceID string
```

Comment:
Since `ResourceID` was used everywhere else (e.g. in resource handler where pods are created/deleted), I wonder if it's better to keep it consistent. Maybe in the future `ResourceID` can be extended to mean other network resources. What do you think?

Reply:
Makes sense. It sounds good to me. Thanks